smartsim/settings/arguments/batch/pbs.py

```diff
@@ -89,15 +97,26 @@ def set_walltime(self, walltime: str) -> None:
         this value will be overridden
 
         :param walltime: wall time
+        :raises ValueError: if walltime format is invalid
+        :raises TypeError: if not type str
         """
-        self.set("walltime", walltime)
+        if not isinstance(walltime, str):
+            raise TypeError("walltime argument was not of type str")
+        pattern = r"^\d{2}:\d{2}:\d{2}$"
+        if walltime and re.match(pattern, walltime):
+            self.set("walltime", walltime)
+        else:
+            raise ValueError("Invalid walltime format. Please use 'HH:MM:SS' format.")
```

Comment:
consider including the value of walltime so the logs help explain what to do to fix the problem.

`Invalid walltime: `{walltime}`. Please use 'HH:MM:SS' format."

Reply:
updated